### PR TITLE
Fix project page issues

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/SWE-335), [Ticket](https://pi-top.atlassian.net/browse/SWE-336) |

#### Main changes
Issue # 1: When deleting a project on bookworm OS, the app crashes.
Fix: Newer versions of PIL have a crop method that will raise an error if the crop boundaries are invalid. Previously, it didn't perform any checks.

Issue # 2: Sometimes projects in the user folders wouldn't be displayed in the menu.
Fix: This happens because the variable for user home directory was generated when the app started. If the systemd service that runs the app starts when there's no display available yet (which is what we use to determine the username and its home folder), this variable would be set to empty, which is why the projects are not found. Now, the user home directory is determined every time the app needs to access or look up the project folders.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: pt_miniscreen/pages/root/projects/utils.py [hunk 1]
```diff
@@ -48,40 +52,55 @@ class ProjectFolderInfo:
...
+            get_folder=lambda: directory,
             title=title,
             exclude_dirs=[],
             recurse_search=False,
             can_remove_all=True,
         )
 
+    @property
+    def folder(self):
+        return self.get_folder()
```
Comment (angusjfw) [inline pt_miniscreen/pages/root/projects/utils.py:64]: good idea

**Review by angusjfw — APPROVED** [review]
